ui/components/Navigator.js

```diff
@@ -591,6 +592,34 @@ class Navigator extends React.Component {
       },
       (err) => console.error(err)
     );
+
+    // Collapse the drawer, on meshmap page to provide more room for the canvas
+    const path = window.location.pathname;
+    if (path.includes("/extension/meshmap")) {
+      this.drawerCollapsedForMeshmap();
+    }
+  }
+
+  // Collapse the drawer, on meshmap page to provide more room for the canvas
+  componentDidUpdate() {
+    const path = window.location.pathname;
+    const { isDrawerCollapsed } = this.props;
+    const { drawerSetForMeshmap } = this.state;
+    if (!drawerSetForMeshmap && !isDrawerCollapsed && path.includes("/extension/meshmap")) {
```

Comment:
`drawerSetForMeshMap` is this variable providing any value to us? I don't see its requirement or am I missing the logic here

Reply:
Yes, there is.
Consider these use cases:
1. The user directly opens the extension page, at that time the whole page would be rendered and  `compomemntDidMount` will be called. So, the drawer will get collapsed.
2. The user reloads the page, then similar situation as above.
3. The user starts at Dashboard i.e, `localhost:9081` and then navigated to extensions page. This time the drawer is already mounted so the `componentDidMount` will not run, but we want the drawer to get collapsed when the user goes to meshmap extension page. So, the `componentDidUpdate` is needed.
4. Now, the user navigates away to some other page and comes back, then we would still want the drawer to collapse.

All the above scenario could be covered without the extra state. But, now conisder the user wants to expand/collapse the drawer manually when on the meshmap extensions page. They won't be able to because on doing so a re-render would happen and the isDrwerCollapsed state would get updated and that would cause the whole collapsing on meshmap page to go on an infinite loop.
So, to prevent that we need another state to keep track that this is the first/returning-first instance of the user to meshmap page and collapse the drawer. After that whatever user does (expand/collapse) on the page is upto them.

A bit long explanation, but exactly printed what I went through while writing the code yesterday 😂

Reply:
hmmm, btw great explanation, that was the reason I was talking about putting it in useEffect hook, I meant the extension code that time. In this code: https://github.com/meshery/meshery/blob/master/ui/components/ExtensionSandbox.js

the drawer-expand state is saved as redux variable, so no prop drilling is required. and the use-case is pretty forward without any extra logic. 
1. load the extension-sandbox component: close the drawer.
2. this removes the overhead from navigator.js to manage any collapse event on any button-click other than that (expnand-collapse) button.

But anyways, this is also good.

Reply:
Ok, that sounds like a better one.
Though I tried that logic and it doesn't seem to work for some reason. Can't seem to figure out the issue, do you mind taking a look?

I have pushed the probable code change for it to the PR, let me know what am I missing here 🤔

Reply:
NVM, fixed the issue. I forgot to pass the toggle to props :p

Everything works fine now.